src/buckets.spec.ts

```diff
@@ -0,0 +1,200 @@
+/* eslint-disable import/first */
+;(global as any).WebSocket = require('isomorphic-ws')
+
+import fs from 'fs'
+import path from 'path'
+import { isBrowser, isNode } from 'browser-or-node'
+import { ThreadID } from '@textile/threads-id'
+import { expect } from 'chai'
+import { Client } from '@textile/threads-client'
+import { InitReply } from '@textile/buckets-grpc/buckets_pb'
+import { Context } from './context'
+import { Buckets } from './buckets'
+import { signUp } from './utils'
+
+// Settings for localhost development and testing
+const addrApiurl = 'http://127.0.0.1:3007'
+const addrGatewayUrl = 'http://127.0.0.1:8006'
+const wrongError = new Error('wrong error!')
+const sessionSecret = 'textilesession'
+
+describe('Buckets...', () => {
+  let ctx: Context = new Context(addrApiurl, undefined)
+  const client = new Buckets(ctx)
+  let buck: InitReply.AsObject
+  let fileSize: number
+  before(async () => {
+    const user = await signUp(ctx, addrGatewayUrl, sessionSecret)
+    ctx = ctx.withSession(user.user?.session).withThreadName('buckets')
+    const id = ThreadID.fromRandom()
+    const db = new Client(ctx)
+    // @todo: Warning, this is a hack!
```

Comment:
@andrewxhill we'll have to update js-threads-client to accept a Context here, otherwise, it just ignores all the extra headers and only uses host and transport. Easy fix, which once done, we can update these tests.

Reply:
sgtm